### PR TITLE
fix: 🐛 handle browsers with limited time zone support

### PR DESCRIPTION
**Pull request by klzns**
[Twist thread](https://twist.com/a/65306/ch/166131/t/3184509/c/73903045)

Safari 13 or macOS Catalina doesn't recognize the time zone `America/Toronto`. This PR checks the availability of the time zone before using.

Before | After
---|---
CleanShot 2022-02-21 at 11 35 19@2x | CleanShot 2022-02-21 at 11 41 14@2x
CleanShot 2022-02-21 at 11 36 50@2x | CleanShot 2022-02-21 at 11 41 26@2x

### CODE REVIEW

**Review by gabrielreisn — APPROVED** [review]

**Comment by gabrielreisn** [comment]
why are you simulating through saucelabs? specific user issue within some old macOS version?

**Comment by klzns** [comment]
@gabrielreisn yes, the user was using Safari 13

> Safari 13 or macOS Catalina doesn't recognize the time zone America/Toronto.

File: src/modules/time.ts [hunk 1]
```diff
@@ -221,7 +223,12 @@ export function formatEpochTime({
   timeFormat: '12-hour' | '24-hour'
   locale: string
 }) {
-  const date = DateTime.fromSeconds(epoch, { zone: timeZone, locale })
+  const isValidTimeZone = Info.isValidIANAZone(timeZone)
+
+  const date = DateTime.fromSeconds(epoch, {
+    zone: isValidTimeZone ? timeZone : undefined,
```
Comment (kaisermann) [inline src/modules/time.ts:229]: What happens if it's not valid?

Comment (klzns) [reply src/modules/time.ts:229]: It uses the local time zone

Comment (kaisermann) [reply src/modules/time.ts:229]: Care to add a comment 👀 ?

**Review by kaisermann — APPROVED** [review]

File: src/modules/time.ts (hunk 1 above)
Comment (DavidPhilip) [reply src/modules/time.ts:229]: @klzns Is this the best possible fallback? Correct me if I'm wrong but if the depot location would always return a non-IANA timezone we'd always use the local one for everything on the team, ETAs etc.

Is it possible to fall back to a valid IANA timezone that has the same offset as the invalid one? Or we use the offset of the invalid one entirely.

Comment (klzns) [reply src/modules/time.ts:229]: @DavidPhilip The time zone is a valid IANA. The issue is that the computer doesn't support all the latest IANA time zones.

**Pull request merged**